File: Other/Google_Foo/answer_new.py
import timeit
def answer(n):
	lambdastring = "2357"
	#if n==0:
	#	lambdastring = str(2)
	i = 11			# check for every number (specifically odd number) for prime
	#count=4		# count of digits of number of prime numbers covered so far

	#generate prime numbers till the digit count is not n
	"""while count < n+5:
		isPrime = True
		for x in range(i-2, 2, -2):
			if i%x == 0:
				isPrime = False
				break
		if isPrime:
			count = count + len(str(i))
		i+=2
		#print(count)

	# resetting the value of count to the length of last prime number
	#if(n > 1):
	i-=2
	lambdastring = lambdastring + str(i)
	i+=2
	#to generate the prime number till the digit count is not 5
	while len(lambdastring) < 5:
		isPrime = True
		for x in range(i-2, 2, -2):
			if i%x == 0:
				isPrime = False
				break
		if isPrime:
			lambdastring = lambdastring + str(i)
			#print(lambdastring)
		#print(lambdastring)
		i+=2
	return lambdastring[0:5]"""

	while len(lambdastring) < n+5:
		isPrime = True
		for x in range(i-2, 2, -2):
			if i%x == 0:
				isPrime = False
				break
		if isPrime:
			lambdastring = lambdastring + str(i)
		i+=2
	return lambdastring[n:n+5]


if __name__ == '__main__':
	a = input("PYTHON: Enter value of n: ")
	start_time = timeit.default_timer()
	print(answer(a))
	elapsed = timeit.default_timer() - start_time
	print("Approx run time: " + str(elapsed * 1000))

File: Other/Google_Foo/Answer_New.java
import java.util.Arrays;
public class Answer_New {   
    public static String answer(int n) { 

        // Your code goes here.
        /*
        * 1. Generate the prime numbers and concat it into
        *    the string till the string length is equal to 
        *    10004 or greater
        *    // think of while lop
        * 2. Return str.substring(n, n+5);
        */
        boolean[] primeArr = new boolean[20234];
        Arrays.fill(primeArr, true);
        //primeArr[0] = false;
        //primeArr[1] = false;
        int x = 4;
        while(x < primeArr.length) {
            primeArr[x] = false;
            x += 2;
        }
        for(int i=3; i<primeArr.length; i++) {
            if(primeArr[i] == true) {
                for(int j=i-1; j>1; j--) {
                    if(i%j == 0) {
                        int k=0;
                        int mul = 0;
                        while(mul < 20234) {
                            primeArr[mul] = false;
                            k++;
                            mul = k*i;
                        }
                        break;
                    }
                }
            }
        }
        StringBuilder lambdaString = new StringBuilder();
        for(int i=2; i<primeArr.length; i++) {
            if(primeArr[i]) {
                lambdaString.append(String.valueOf(i));
            }
        }
        return lambdaString.substring(n, n+5);
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();
        System.out.println(Answer_New.answer(944));
        long endTime   = System.currentTimeMillis();
        long totalTime = endTime - startTime;
        System.out.println("Approx run time: "+totalTime);
    }
}